### PR TITLE
Added rev_history topic template, and fixed replaceable markup in gui…

### PR DESCRIPTION
**Pull request by Bilhar-rh**
This PR adds the rev_history template topic, and also fixes markup with replaceables in command syntax.

@adellape @bfallonf @tpoitras @tnguyen-rh PTAL and let me know if anything needs to be changed before I merge.

### CODE REVIEW

**Comment by tnguyen-rh** [comment]
It looks like the build failed due to lacking rev_history directory.
Am i missing something?

File: _build_cfg.yml [hunk 1]
```diff
@@ -118,7 +118,9 @@ Topics:
         File: throttling
       - Name: Source Control Management
         File: scm
-
+  - Name: Revision History
+    File: rev_history
```
Comment (adellape) [inline _build_cfg.yml]: I'm guessing these are vestigial from an earlier local test run, causing the current build failure.

**Comment by adellape** [comment]
@CowboysFan While you're in the file again, could you comb through the "Code blocks" and "Inline Code or Commands" sections to make sure they jive with the latest (specifically related to replaceables). I think there's some remaining language that may be out of date.

Also, this anchor link in "Inline Code or Commands" doesn't work, at least in GH:

```
See <<Command syntax and examples>> for information on how to show commands and command syntax.
```

**Comment by CowboysFan** [comment]
@adellape Thanks, I'll take a look. And yes, the build's broken because I was testing some local builds and forgot to revert the config file back. I'll fix that with my next pass through.

**Comment by CowboysFan** [comment]
@adellape @tnguyen-rh @tpoitras @bfallonf Okay, I've made some updates and reverted the build file back. It's building locally, but Travis CI seems to be stuck. Let me know if anything else needs to be changed before I merge it.

File: contributing_to_docs/doc_guidelines.adoc [hunk 2]
```diff
@@ -215,15 +215,7 @@ For all of the system blocks including table delimiters, use four characters. Fo
...
-Code blocks are used to show examples of command screen outputs, or configuration files. When using command blocks always use the actual values for any items that a user would normally replace. Code blocks should represent exactly what a customer would see on their screen. If you need to expand or provide information on what some of the contents of a screen output or configuration file represent, then use callouts to provide that information.
-
-Follow these general guidelines when using code blocks:
-
-* Do NOT show replaceables within code blocks.
-
-* Do NOT use any markup in code blocks; code blocks generally do not accept any markup
-
-* Try to use callouts to provide information on what the output represents when required
+Code blocks are used to command usage, syntax, and examples of command screen outputs or configuration files.
```
Comment (adellape) [inline contributing_to_docs/doc_guidelines.adoc]: s/used to/used to show/ ?

File: contributing_to_docs/doc_guidelines.adoc [hunk 3]
```diff
@@ -248,21 +240,20 @@ $ lorem.sh
 
 Without the line spaces the content is likely to be not parsed correctly.
 
-=== Inline Code or Commands
-Do NOT show full commands or command syntax inline within a sentence. See <<Command syntax and examples>> for information on how to show commands and command syntax.
+=== Command syntax and examples
+Code blocks are used to show command usage, syntax, and examples of command screen outputs or configuration files. Code blocks can be used in a few different ways:
```
Comment (adellape) [inline contributing_to_docs/doc_guidelines.adoc]: It seems weird to start both of these back to back sections with the same sentence (line 218 for "Code blocks").

File: contributing_to_docs/doc_guidelines.adoc [hunk 4]
```diff
@@ -248,21 +240,20 @@ $ lorem.sh
...
+Follow these general guidelines when using code blocks:
 
-Use the `GET` operation to do x.
+* Always show command syntax or usage with any parameters or values as <replaceables>
 
-=== Command syntax and examples
-The main distinction between showing command syntax and example is that a command syntax should just show customers how to use the command without real values. An example on the other hand should show the command with actual values with an example output of that command, where applicable.
+* If the command or command syntax warrants a full example, wrap it inside an example block, and use actual example values. Do NOT use <replaceables> in command example outputs.
+
+* Try to use callouts to provide information on what the output represents when required
```
Comment (adellape) [inline contributing_to_docs/doc_guidelines.adoc]: We mention callouts here and in another spot below, yet we don't explain anywhere what they are or show how to use them.

File: contributing_to_docs/templates/topic_template.adoc [hunk 5]
```diff
@@ -12,4 +12,4 @@ toc::[]
 == Overview
 --All topics should include an Overview section that introduces the topic. Remove this and add a brief introduction here.--
 
-== Section Title
+== Section Title
```
Comment (tnguyen-rh) [inline contributing_to_docs/templates/topic_template.adoc]: Probably removing the newline from eof was a typo?
It seems asciidoctor doesn't care, but for future-proofing against lame tools, we probably want to maintain eof newline.
Reverting this hunk also makes for a smaller diff, always a good thing.  :)

**Comment by bfallonf** [comment]
@CowboysFan remember to fix the "Quick markup reference" table so that it doesn't talk about sidebar blocks in the replaceables section.

File: contributing_to_docs/doc_guidelines.adoc (hunk 4 above)
Comment (bfallonf) [inline contributing_to_docs/doc_guidelines.adoc]: @CowboysFan I'd suggest this "Command Syntax and Examples" section above be condensed a little. We've got two lists (one using numbers, and the other dot points) that are for the same purpose. The list items are even quite similar. I'd suggest that the first (numbered) list is redundant, as the info there is either in the little intro para, or the dot point list.

**Comment by adellape** [comment]
@CowboysFan Some outstanding comments here.

**Comment by adellape** [comment]
@CowboysFan This needs a rebase now.

**Comment by CowboysFan** [comment]
@adellape Yeah, I realize that. Got some other higher priority items on the list, but I'll definitely rebase when I fix all this up.

**Comment by CowboysFan** [comment]
@adellape @bfallonf @tnguyen-rh Sorry, I got sidetracked with other things. I'll try to get all this fixed up and update this PR. Apologies for the delay.

**Comment by CowboysFan** [comment]
@adellape @bfallonf @tnguyen-rh @tpoitras Please take a look and see if I've covered everything from these comments. I hope I addressed all the issues, but let me know I still missed something.

**Comment by bfallonf** [comment]
@CowboysFan This looks good to me.

File: contributing_to_docs/doc_guidelines.adoc [hunk 6]
```diff
@@ -421,24 +434,22 @@ The `GET` operation can be used to do something.
...
-a|$$`*ENVIRONMENT_VARIABLE*`$$
+a|$$*`ENVIRONMENT_VARIABLE`*$$
 
-$$`*PARAMETER*`$$
-a|Use the `*IP_ADDRESS*` environment variable for the server IP address.
+$$*`PARAMETER`*$$
+a|Use the *`IP_ADDRESS*` environment variable for the server IP address.
 
-The `*MAX_PODS*` parameter limits the number of pods you can have.
+The *`MAX_PODS`* parameter limits the number of pods you can have.
```
Comment (adellape) [inline contributing_to_docs/doc_guidelines.adoc]: These are now backwards. It was correct the way it was (and how we've been using it). From http://asciidoctor.org/docs/user-manual/#mono:

"Monospaced text can be bold and italicized, as long as the markup sets are entered in the right order. The monospace markup must be the outermost set, then the bold set, and the italic markup must always be the innermost set."

**Comment by CowboysFan** [comment]
@adellape True, but I think this way it works both in GitHub and with the build mechanism. This was an issue brought up where the rendering is not correct in GitHub. I think this way we won't have any issues, but let me know if you guys think it breaks something.

**Comment by adellape** [comment]
@CowboysFan So, I did verify that the recommended syntax...

```
`*MAX_PODS*`
```

and the proposed change syntax...

```
*`MAX_PODS`*
```
...do both seem to render correctly (i.e., bold and monospaced) in local builds. Granted, this was just after some fairly simple testing, could definitely be some edge cases I'm not thinking of. And in GitHub's rendered adoc previews, the recommended syntax appears as monospaced with visible asterisks around it, while the proposed syntax appears as bold monospace. 

I'm still hesitant about this change, though. Comparing the pros/cons/unknowns of the proposed change:

Pros:
- GitHub rendered adoc previews look slightly nicer. This seems a bit moot though, considering we can't rely on GH's previews anyway since we use ifdefs. E.g., a heavily-conditionalized topic such as https://github.com/openshift/openshift-docs/blob/master/admin_guide/install/quick_install.adoc is pretty much useless in GH previews, and that's just one of many examples.

Cons: 
- Inconsistent with standard practices (http://asciidoctor.org/docs/user-manual/#mono as mentioned previously).
- Would need to safely change the 400+ instances of the previous usage and verify the changes.
- Contributors/devs by now seem to have gotten used to the existing style (based on standard practice) and would have to learn/remember to switch usage going forward (and/or we'd have to correct).
- Contributors coming in fresh would have to be made aware of this quirk instead of being able to rely on standard practice.

Unknowns:
- How do Customer Portal builds behave with this change? Has anyone tested it? I would hate to make this switch, only to later find out that it's yet another thing that the Portal builds don't like.
- Potentially impacts the portability of our source, as far as how other future process may behave. Sticking with the recommended syntax at least hedges our bets a bit.

The sticking points for me are that we'd be deviating from the recommended practice seemingly just to make a low-priority (in my opinion) render look slightly nicer. I say low-priority since the first-class builds that we push people towards are the docs.openshift.* sites and the Customer Portal builds, plus the GH previews are unreliable anyway (per above mentioned ifdefs) and I therefore honestly put little stock in them. And an important question we should at the very least answer/verify is how the change looks on the Portal.

cc @tnguyen-rh @tpoitras @bfallonf @vikram-redhat

**Comment by CowboysFan** [comment]
@adellape Okay, I'll leave these as-is and not worry about how they render in GitHub. That way we stay with the standard AsciiDoc practices.

**Comment by adellape** [comment]
:+1: Sounds good. Thanks.

**Comment by CowboysFan** [comment]
@adellape @tpoitras @bfallonf @tnguyen-rh @vikram-redhat Unless there is something else  missing, I think this is ready to be merged. Thanks.

**Comment by adellape** [comment]
LGTM

**Pull request merged**